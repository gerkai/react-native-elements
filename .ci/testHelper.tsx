import React from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider, FullTheme } from '../src/config';

const theme = {
  colors: {
    black: 'black',
  },
};

export const renderWithTheme = (
  children: any,
  themeProp: Partial<FullTheme> = theme
) => {
  return render(<ThemeProvider theme={themeProp}>{children}</ThemeProvider>);
};
