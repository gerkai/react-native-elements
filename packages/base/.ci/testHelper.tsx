import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';
import { ThemeProvider, FullTheme, colors } from '../src/config';
import deepmerge from 'deepmerge';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  themeProp: Partial<FullTheme> = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(
    themeProp ? (
      <ThemeProvider theme={deepmerge({ colors }, themeProp)}>
        {children}
      </ThemeProvider>
    ) : (
      children
    ),
    options
  );
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper');
  return { wrapper, ...renderApi };
};
