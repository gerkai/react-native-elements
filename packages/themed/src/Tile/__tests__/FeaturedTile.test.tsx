import React from 'react';
import { FeaturedTile } from '..';
import { renderWithWrapper } from '../../../.ci/testHelper';

describe('FeaturedTitle component', () => {
  it.skip('should apply values from theme', () => {
    const theme = {
      FeaturedTile: {
        title: 'I am featured',
      },
    };
    const { queryByText } = renderWithWrapper(
      <FeaturedTile featured imageSrc={{ uri: 'http://google.com' }} />,
      '',
      theme
    );
    expect(queryByText('I am featured')).not.toBeNull();
  });
});
