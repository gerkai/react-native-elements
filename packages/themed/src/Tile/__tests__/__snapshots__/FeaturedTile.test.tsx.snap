// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedTitle component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    replaceTheme={[Function]}
    style={
      Object {
        "height": 600,
        "width": 750,
      }
    }
    theme={
      Object {
        "colors": Object {
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "searchBg": "#dcdce1",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "default": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "ios": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "web": Object {
              "error": "#ff190c",
              "grey": "#393e42",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#ca71eb",
          "success": "#52c41a",
          "warning": "#faad14",
          "white": "#ffffff",
        },
      }
    }
    updateTheme={[Function]}
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "alignItems": "center",
          "height": 600,
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": "http://google.com",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 600,
              "width": 750,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "bottom": 0,
            "flex": 1,
            "justifyContent": "center",
            "left": 0,
            "paddingBottom": 40,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 45,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "justifyContent": "center",
            }
          }
        />
        <Text
          accessibilityRole="text"
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0)",
              "color": "#ffffff",
              "fontSize": 28.285714285714285,
              "marginBottom": 15,
              "textAlign": "center",
            }
          }
          testID="featuredTileTitle"
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
